src/document.c

```diff
@@ -1955,37 +1961,51 @@ static gchar *write_data_to_disk(const gchar *locale_filename,
 		/* Use POSIX API for unsafe saving (GVFS-unsafe) */
 		/* The error handling is taken from glib-2.26.0 gfileutils.c */
 		errno = 0;
-		fp = g_fopen(locale_filename, "wb");
+		fp = g_fopen(locale_filename, "w+b"); // we can not destroy  symbolic/hard links
```

Comment:
We need open for writing without truncation to empty,  set write position to start, truncate to dataLength,  write data to file.